merino-settings/src/providers.rs

```diff
@@ -215,6 +217,60 @@ impl Default for StealthConfig {
     }
 }
 
+/// Settings for Merino suggestion providers.
+#[derive(Serialize, Deserialize)]
+pub struct SuggestionProviderSettings(pub HashMap<String, SuggestionProviderConfig>);
+
+impl SuggestionProviderSettings {
+    /// Load settings for suggestions providers.
+    ///
+    /// The organization of the provider configuration files is identical to the
+    /// top level settings.
+    ///
+    /// Note that settings for suggestion providers cannot be configured via
+    /// environment variables.
+    ///
+    /// # Errors
+    /// If any of the configured values are invalid, or if any of the required
+    /// configuration files are missing.
+    pub fn load() -> Result<Self> {
+        let mut s = Config::new();
+
+        // Start off with the base config.
+        s.merge(File::with_name("./config/providers/base"))
+            .context("loading base config for suggestion providers")?;
+
+        // Merge in an environment specific config.
+        let merino_env = std::env::var("MERINO_ENV").unwrap_or_else(|_| "development".to_string());
+        s.merge(File::with_name(&format!("./config/providers/{}", merino_env)).required(false))
+            .context("loading environment config for suggestion providers")?;
+
+        // Add a local configuration file that is `.gitignore`ed.
+        s.merge(File::with_name("./config/providers/local").required(false))
+            .context("loading local config for suggestion providers")?;
+
+        serde_path_to_error::deserialize(s)
+            .map(SuggestionProviderSettings)
```

Comment:
This seems to work around the deserialization error, though I am still not entirely sure why the direct deserialization didn't work in the first place.